storage/__init__.py

```diff
@@ -76,31 +76,36 @@ def regenerate(self):
         """Generate an auth token for the selected identity."""
         # If the server_url has a hash fragment, it's a storage node and
         # that's the secret.  Otherwise it's a token server url.
+        uid = self.uid
         url = urlparse(self.server_url)
         if url.fragment:
-            endpoint = url._replace(fragment="", path="/1.5/" + str(self.uid))
+            endpoint = url._replace(
+                path=url.path.rstrip("/") + "/1.5/" + str(uid),
```

Comment:
If you want to run this against a local server via https://github.com/mozilla-services/syncserver, it needs to support a `/storage` path prefix.